tests/client/executor_client.h

```diff
@@ -1,10 +1,13 @@
 #ifndef EXEC_CLIENT_H
 #define EXEC_CLIENT_H
 
-#include <signal.h>
-#include <stdlib.h>
-#include <sys/wait.h>
-#include "../../runtime_util/runtime_util.h"
+#include <errno.h>     // for errno
+#include <signal.h>    // for SIGINT
+#include <stdio.h>     // for printf, stderr
+#include <stdlib.h>    // for malloc, free, realloc, exit
+#include <string.h>    // for strcat, strcpy
+#include <sys/wait.h>  // for waitpid, pid_t
+#include <unistd.h>    // for execlp, fork, chdir
```

Comment:
Not sure if this was autofill from VSCode but I liked it the previous way (although the comments are useful). The reason for the `runtime_util` include is because it's meant to be a place where a bunch of headers that are commonly included are included, and the reader can know like "oh, he included that file, which I know contains such-and-such headers".....

Reply:
Imo you shouldn't include `runtime_util` just for headers as it masks what is really needed in the source file. One thing I hate about C/C++ is you don't know where a function/type is defined just by looking at the source file. You should only include `runtime_util` if you need some of the functions/types specific to Runtime defined in it.

Reply:
Ok that's a fair point, just the opposite of what I've been doing since high school though so if you want to change that we should keep it consistent and do it for all of Runtime :\\
Are you saying that we should not include any headers in `runtime_util` (or only the standard library headers that are actually used in `runtime_util`) and for each file include things that are necessary in that file only?

Reply:
uhh, I don't care too much about that, though that would probably be simpler. I just think you shouldn't import a file just for its headers. In the rest of the project, we only import `runtime_util` if we need its attributes